include/f_core/util/linear_fit.hpp

```diff
@@ -0,0 +1,73 @@
+#include "f_core/util/circular_buffer.hpp"
+
+#include <array>
+#include <cstddef>
+
+template <typename T, std::size_t len>
+class CRollingSum {
+  public:
+    static_assert(len > 0, "What is the sum of 0 elements? You probably don't want this (also it will break)");
+
+    using value_type = T;
+    static constexpr std::size_t size_ = len;
+
+    constexpr CRollingSum(T start) : buf(start) { fill(start); }
+
+    constexpr std::size_t size() const { return size_; }
+
+    constexpr void fill(const value_type &start) {
+        buf.Fill(start);
+        total = buf.OldestSample();
+        for (std::size_t i = 1; i < buf.Size(); i++) {
+            total = total + buf[1];
+        }
```

Comment:
buf[1] or buf[i]?

Reply:
technically forall i, buf[i] = buf[1] in this case because of the Fill call above but saying buf[i] makes much more sense. 
Though theyre the same, I kept it as a loop to rather than multiplication attempt to safeguard against any funny floating point weirdness. Using the loop here, Feed() vs. Fill() do their calculations the same way.